### PR TITLE
add FAQ for divergent changes

### PR DESCRIPTION
**Pull request by tp-woven**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: docs/FAQ.md
Comment (PhilipMetzger) [inline docs/FAQ.md:112]: I would also mention divergent branches here, as they also belong in the same bucket. Here's my suggestion:
> What do I do with divergent changes and branches?

File: docs/FAQ.md
Comment (PhilipMetzger) [inline docs/FAQ.md:116]: Here a sentence on how to resolve divergent branches would also be nice. Here's my suggesion for it, `For divergent branches, setting the branch to the correct version via "jj branch set" should resolve the divergence. As repeatedly abandoning commits does not resolve the divergence.`

**Review by PhilipMetzger — APPROVED** [review]
LG, but it should also mention divergent branches.

File: docs/FAQ.md
Comment (martinvonz) [reply docs/FAQ.md:112]: Speaking of that, should we start calling them "conflicted changes" instead "divergent changes"? We call branches "conflicted", not "divergent". We also use "conflicted" for files. I don't think we need to change the terminology before this PR gets merged.

File: docs/FAQ.md
Comment (martinvonz) [reply docs/FAQ.md:116]: nit: drop the "As" there (or join the sentences with ", as"). More importantly, I'm not sure we should even include that last sentence. First, it's not actually true - repeatedly abandoning commits will resolve the divergence, once the one side becomes an ancestor of the other. Do you think it's a common thing to think that repeatedly abandoning commits would be a good strategy for resolving conflicted branches? If not, it seems better to not mention it at all.

File: docs/FAQ.md
Comment (PhilipMetzger) [reply docs/FAQ.md:112]: I think we should call them "conflicted changes" for user-facing documentation, while keeping "divergence" for technical documentation.

File: docs/FAQ.md
Comment (PhilipMetzger) [reply docs/FAQ.md:116]: > Do you think it's a common thing to think that repeatedly abandoning commits would be a good strategy for resolving conflicted branches? If not, it seems better to not mention it at all.

I've mentioned it as a counter example, as that was my experience from the last time we talked about it in [Discord].

[Discord]: https://discord.com/channels/968932220549103686/969829516539228222/1144289598579683429

Comment (martinvonz) [reply docs/FAQ.md:116]: I can see how thinking that abandoning one side would help, but I don't think users are going to think that repeatedly abandoning commits will help. So I'm fine with something like "Note that abandoning one side of the conflicted branch does not help - it just moves that side of the branch to the parent commit.".

File: docs/FAQ.md
Comment (ilyagr) [reply docs/FAQ.md:112]: I think that if we use "conflicted", it should be "conflicted change ids", since it's unclear what a "change" is and whether it's different from "revision" (in my mind, these are the same or very close). I use "divergent changes" to refer to "revisions with conflicted change ids".

File: docs/FAQ.md
Comment (ilyagr) [reply docs/FAQ.md:116]: Perhaps this should be a separate question? We can also link to the Glossary, were I think Martin carefully explained the difference and similarity between conflicted branches and change ids.

Alternatively, we could make it a separate paragraph in this question, but probably at the end.

File: docs/FAQ.md
Comment (ilyagr) [inline docs/FAQ.md:115]: I would mention that these commits need to be referred to the commit id.

> To refer to such commits, you must use their [commit id](link to glossary). Most commonly ... abandon the unneeded commits (using their commit ids).

File: docs/FAQ.md
Comment (ilyagr) [reply docs/FAQ.md:112]: So, I'd title this:

> How do I deal with conflicted change ids (?? after the id), AKA divergent revisions?

File: docs/FAQ.md
Comment (martinvonz) [reply docs/FAQ.md:116]: Yes, I think it's makes sense to make it a separate question. It would be easier to link to that way. And it also doesn't block this PR :)

File: docs/FAQ.md
Comment (ilyagr) [reply docs/FAQ.md:115]: Considering https://github.com/martinvonz/jj/discussions/2485, we could also say something like

> If you would like to keep both versions of the revision, you can `jj duplicate` one of them before the `jj abandon`.

File: docs/FAQ.md
Comment (tp-woven) [reply docs/FAQ.md:112]: I don't mind changing to conflicted changes / change IDs if there's consensus on it but personally, as a user, they feel like two different things to me, and using the same "name" to describe them would make it more confusing for me.

File: docs/FAQ.md
Comment (tp-woven) [reply docs/FAQ.md:116]: OK, I'll add this as a separate question 👍

File: docs/FAQ.md
Comment (martinvonz) [reply docs/FAQ.md:112]: I think it makes sense to use the same name because branches and change ids behave quite similarly, but I don't want that discussion to block this PR. If you put the question about conflicted branches in a separate question, then the different terminology becomes less apparent.

File: docs/FAQ.md
Comment (martinvonz) [inline docs/FAQ.md]: nit: There are two question marks (since 65de7cb0eb0c06750663b7db24830be4ee30520f and cd4d6c9bb19f89b5a0a87e2c48dff8f0511fec3f)

File: docs/FAQ.md
Comment (tp-woven) [reply docs/FAQ.md]: Done.

File: docs/FAQ.md
Comment (tp-woven) [reply docs/FAQ.md]: OK, I updated the PR with a separate question for branches.

I agree about branches and change IDs using the same terminology[1], but I still prefer divergent (or some other term that isn't conflict) to represent that, since to me "conflicted change/branch" makes me think it's a change/commit that has a file conflict in it. But as you say, that's a separate discussion :) 

[1] Though I would argue that they should also be represented the same way in the log

**Comment by tp-woven** [comment]
Thanks for all the comments and suggestions on the PR!

(I think) I addressed all of them, so I'll go ahead and merge, but please let me know if there's anything else you'd like me to change and I can follow up in a separate PR.

File: docs/FAQ.md
Comment (martinvonz) [inline docs/FAQ.md:135]: `| grep A 2 <branch>` is not guaranteed to include all targets. Maybe this is still a good-enough approximation for now. We should probably allow `jj branch list <branch>`.

**Review by martinvonz — APPROVED** [review]

File: docs/FAQ.md
Comment (tp-woven) [reply docs/FAQ.md:135]: Yeah, I can change it to a larger number, I suppose, but `jj branch list <branch>` would be really good (and/or `jj branch list --conflicted`).

Comment (ilyagr) [reply docs/FAQ.md:135]: (Optional) I'd just say "Use `jj branch list` to see all the commits associated with a branch". The `grep` trick is nice, but this doesn't seem like the right place for it.

> We should probably allow jj branch list <branch>.

+1.

File: docs/FAQ.md
Comment (ilyagr) [inline docs/FAQ.md:129]: (Optional) Instead of "that can't be resolved automatically", we could write "because `jj` couldn't fully resolve its desired position"

File: docs/FAQ.md
Comment (ilyagr) [inline docs/FAQ.md:115]: I still feel that the word "change id" should be somewhere in here, e.g. "... associated with a single change id".

File: docs/FAQ.md
Comment (ilyagr) [inline docs/FAQ.md:118]: (Optional) Instead of "both revisions of the change", I'd say "both versions of the commit". As I said before, I find the word "change" a bit ambiguous on its own.

File: docs/FAQ.md
Comment (ilyagr) [reply docs/FAQ.md:115]: If this was already discussed and rejected, it doesn't have to be blocking.

File: docs/FAQ.md
Comment (tp-woven) [reply docs/FAQ.md]: Oh, didn't realize. That's great!

File: docs/FAQ.md
Comment (tp-woven) [reply docs/FAQ.md:135]: Fair enough, let me remove the grep, people can figure it out on their own.

File: docs/FAQ.md
Comment (tp-woven) [reply docs/FAQ.md:115]: Ah, sorry, missed that. Let me add it somewhere.

File: docs/FAQ.md
Comment (tp-woven) [reply docs/FAQ.md:118]: Thanks. I think I'll use "both commits with this change id". "Both versions of the commit" sounds more confusing to me (in my mind, they are two separate commits, not versions of the same commit 😅)

**Pull request merged**
